tests/appsec/waf/test_telemetry.py

```diff
@@ -1,28 +1,224 @@
-from utils import interfaces, released, rfc, weblog, scenarios, context
+from utils import interfaces, released, rfc, weblog, scenarios, context, bug, irrelevant, missing_feature
 
 
 TELEMETRY_REQUEST_TYPE_GENERATE_METRICS = "generate-metrics"
 TELEMETRY_REQUEST_TYPE_DISTRIBUTIONS = "distributions"
 
 
-def _validate_headers(headers, request_type):
-    """https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/how-to-use.md
+def _setup(self):
+    """
+    Common setup for all tests in this module. They all depend on the same set
+    of requests, which must be run only once.
     """
+    # Run only once, even across multiple class instances.
+    if hasattr(Test_TelemetryMetrics, "__common_setup_done"):
+        return
+    r_plain = weblog.get("/", headers={"x-forwarded-for": "80.80.80.80"})
+    r_triggered = weblog.get("/", headers={"x-forwarded-for": "80.80.80.80", "user-agent": "Arachni/v1"})
+    r_blocked = weblog.get(
+        "/",
+        headers={"x-forwarded-for": "80.80.80.80", "user-agent": "dd-test-scanner-log-block"},
+        # XXX: hack to prevent rid inhibiting the dd-test-scanner-log-block rule
+        rid_in_user_agent=False,
+    )
+    Test_TelemetryMetrics.__common_setup_done = True
+
+
+@rfc("https://docs.google.com/document/d/1qBDsS_ZKeov226CPx2DneolxaARd66hUJJ5Lh9wjhlE")
+@released(python="?", cpp="?", golang="?", java="1.12.0", dotnet="?", nodejs="?", php="?", ruby="?")
+@scenarios.appsec_waf_telemetry
+class Test_TelemetryMetrics:
+    """Test instrumentation telemetry metrics, type of metrics generate-metrics"""
+
+    setup_all_telemetry_requests_are_successful = _setup
+
+    def test_all_telemetry_requests_are_successful(self):
+        """Tests that all telemetry requests succeed."""
+        for data in interfaces.library.get_telemetry_data():
+            assert data["response"]["status_code"] == 202
+
+    setup_headers_are_correct = _setup
+
+    @bug(context.library < "java@1.13.0", reason="Missing two headers")
+    @missing_feature(context.weblog_variant == "spring-boot-native", reason="GraalVM. Tracing support only")
```

Comment:
maybe putting this decorator at class level ?

Reply:
Done.